azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/WebAppDeploymentSlot.java

```diff
@@ -34,81 +39,163 @@ public WebAppDeploymentSlot(WebAppDeploymentSlotEntity deploymentSlot, AzureAppS
 
     @Override
     public IWebApp webApp() {
-        return null;
+        final WebAppDeploymentSlotEntity entity = entity();
+        return azureAppService.webapp(entity.getResourceGroup(), entity.getWebappName());
     }
 
     @Override
     public IWebAppDeploymentSlotCreator create() {
-        return null;
-    }
-
-    @Override
-    public WebAppDeploymentSlotEntity entity() {
-        return null;
+        return new WebAppDeploymentSlotCreator();
     }
 
     @Override
     public void start() {
-
+        getDeploymentSlotClient().start();
     }
 
     @Override
     public void stop() {
-
+        getDeploymentSlotClient().stop();
     }
 
     @Override
     public void restart() {
-
+        getDeploymentSlotClient().restart();
     }
 
     @Override
     public void delete() {
-
+        getDeploymentSlotClient().parent().deploymentSlots().deleteByName(slotEntity.getName());
     }
 
     @Override
     public void deploy(File file) {
-
+        deploy(AppServiceUtils.getDeployTypeByFileExtension(file), file);
     }
 
     @Override
     public void deploy(DeployType deployType, File file) {
-
+        getDeploymentSlotClient().deploy(com.azure.resourcemanager.appservice.models.DeployType.fromString(deployType.getValue()), file);
     }
 
     @Override
     public boolean exists() {
-        return false;
+        return getDeploymentSlotClient(true) != null;
     }
 
     @Override
     public String hostName() {
-        return null;
+        return getDeploymentSlotClient().defaultHostname();
     }
 
     @Override
     public String state() {
-        return null;
+        return getDeploymentSlotClient().state();
     }
 
     @Override
     public Runtime getRuntime() {
-        return null;
+        return AppServiceUtils.getRuntimeFromWebApp(getDeploymentSlotClient());
     }
 
     @Override
     public PublishingProfile getPublishingProfile() {
-        return null;
+        return AppServiceUtils.getPublishingProfile(getDeploymentSlotClient().getPublishingProfile());
+    }
+
+    @Override
+    public WebAppDeploymentSlotEntity entity() {
+        return slotEntity;
+    }
+
+    private com.azure.resourcemanager.appservice.models.DeploymentSlot getDeploymentSlotClient() {
+        return getDeploymentSlotClient(false);
```

Comment:
I don't like the getXXX method with force parameter

Reply:
Look asc for a better name

Reply:
change original force get method to refresh